homeassistant/components/velbus/sensor.py

```diff
@@ -50,58 +50,36 @@ def __init__(
         """Initialize a sensor Velbus entity."""
         super().__init__(channel)
         self._is_counter: bool = counter
-
-    @property
-    def unique_id(self) -> str:
-        """Return unique ID for counter sensors."""
-        unique_id = super().unique_id
+        # define the unique id
         if self._is_counter:
-            unique_id = f"{unique_id}-counter"
-        return unique_id
-
-    @property
-    def name(self) -> str:
-        """Return the name for the sensor."""
-        name = super().name
+            self._attr_unique_id = f"{self._attr_unique_id}-counter"
+        # define the name
         if self._is_counter:
-            name = f"{name}-counter"
-        return name
-
-    @property
-    def device_class(self) -> str | None:
-        """Return the device class of the sensor."""
+            self._attr_name = f"{self._attr_name}-counter"
+        # define the device class
+        if self._is_counter:
+            self._attr_device_class = DEVICE_CLASS_ENERGY
+        elif channel.is_counter_channel():
+            self._attr_device_class = DEVICE_CLASS_POWER
+        elif channel.is_temperature():
+            self._attr_device_class = DEVICE_CLASS_TEMPERATURE
+        # define the icon
+        if self._is_counter:
```

Comment:
We don't need six checks that are the same. Please refactor this. I suggest introducing entity descriptions to do this declaratively.